### DISCUSSION TITLE
[Kubernetes] - Is creating a single managed Kubernetes secret which includes all Infisical Secrets safe?

### DISCUSSION THREAD

**Discussion by 2snEM6** [Infisical/infisical#635] [Q&A] [answered]
While setting up Infisical, I noticed that instead of creating one Kubernetes Managed Secret per Infisical Secret, it creates one single managed secret which includes all Infisical Secrets as its keys.

For example:

In Infisical there are `Key A` and `Key B`.

The operator creates as many Kubernetes keys as declared in the Yaml files, but it includes `Key A` and `Key B` in all of them.

Am I doing something wrong?

**Comment by maidul98** [comment]
Hello @2snEM6 you are correct. The managed secret will fetch in all of the secrets for the given project the token has access to. This is the expected behavior. Were you looking to only fetch one single secret from a project?

**Reply by 2snEM6** [reply]
That is correct. I understand that creating and maintaining multiple projects can be tedious,specially when there are many microservices in the infrastructure.

My idea was to have a single project create N secrets per key entry, but I can see now how that can be limiting, specially for multikey secrets.

**Reply by maidul98** [reply, marked as answer]
I think you should look into our folder features. This will help you group related microservices into one project. You can then generate multiple service tokens for the given folders you placed your secrets into

**Reply by 2snEM6** [reply]
Will do! thank you for your help!
